### DISCUSSION TITLE
[ZK Stack] - Nix package manager previously worked, but now it stopped working

### DISCUSSION THREAD

**Discussion by TicTacWeb3** [zkSync-Community-Hub/zksync-developers#495] [ZK Stack] [answered]
### Module Affected

- [ ] System contracts (era-system-contracts)
- [ ] Smart contracts (era-contracts)
- [ ] zkEVM (era-zk_evm)
- [X] Server (zksync-era)
- [ ] Prover (era-boojum)
- [ ] Fee withdrawer (era-fee-withdrawer)
- [ ] Block Explorer (clock-explorer)
- [ ] Portal - Wallet + Bridge (dapp-portal)
- [X] Other (please specify below)

### Rust Version

rustc 1.77.2 (25ef9e3d8 2024-04-09)

### Operating System

- [ ] Ubuntu
- [ ] MacOS
- [ ] Windows
- [X] Other (Please specify below)

### Issue Description

on EndevourOS, an Arch-Based distribution, when i try to configure my enviroment with the nix package manager, it fails 

### Expected Behavior

the expected behavior is to log me in the nix shell after the completion of the configuration

### Current Behavior

the currect behaviour gives me this error: 
warning: Git tree '/home/endre/Scrivania/SyntraNet/zksync-era' is dirty
error:
       … while calling the 'derivationStrict' builtin

         at /builtin/derivation.nix:9:12: (source not available)

       … while evaluating derivation 'nix-shell'
         whose name attribute is located at /nix/store/a763lrpz1xymjm9bddh6hagzyq01xkz8-source/pkgs/stdenv/generic/make-derivation.nix:348:7

       … while evaluating attribute 'buildInputs' of derivation 'nix-shell'

         at /nix/store/a763lrpz1xymjm9bddh6hagzyq01xkz8-source/pkgs/stdenv/generic/make-derivation.nix:395:7:

          394|       depsHostHost                = elemAt (elemAt dependencies 1) 0;
          395|       buildInputs                 = elemAt (elemAt dependencies 1) 1;
             |       ^
          396|       depsTargetTarget            = elemAt (elemAt dependencies 2) 0;

       (stack trace truncated; use '--show-trace' to show the full trace)

       error: lib.fileset.unions: Element 4 (/nix/store/l0jd9g4nh549c3kpqd8g11crz7gbx84h-source/sdk) is a path that does not exist.
           To create a file set from a path that may not exist, use `lib.fileset.maybeMissing`.

### Prior Research

- [X] Yes, I searched for similar issues and did not find any matching ones.

**Comment by bxpana** [comment]
Will check things out and get back to you.

**Reply by TicTacWeb3** [reply, marked as answer]
with the new update now works, thanks to all the team for correcting fast :)

**Reply by bxpana** [reply]
That's great to hear! Glad it's working for you now.
